File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/valutazioni/ValutazioneTestata.cs
using AgronicaCoreModelsSTD.baseClass;
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.valutazioni
{
    public class Valutazione_Testata
    {
        public PK primaryKey { get; set; }

        public string Ragione_Sociale { get; set; }
        public int Valutazione_Piano_Cod { get; set; }
        public string Valutazione_Piano_Des { get; set; }
        public string Data_Redazione { get; set; }
        public string Note { get; set; }
        public List<Valutazione_TestataxAnno> Valutazione_TestataxAnno { get; set; }

        public bool flag_cancellazione { get; set; }


        public Valutazione_Testata()
        {
            this.flag_cancellazione = false;
            Valutazione_TestataxAnno = new List<Valutazione_TestataxAnno>();
        }

        public Valutazione_Testata(PK primaryKey)
        {
            this.primaryKey = primaryKey;
            this.flag_cancellazione = false;
            Valutazione_TestataxAnno = new List<Valutazione_TestataxAnno>();
        }

        public class PK
        {
            public string Piva { get; set; }
            public int Id_Testata { get; set; }

            public PK(string piva, int codice)
            {
                this.Piva = piva;
                this.Id_Testata = codice;
            }

            public PK()
            {
                this.Piva = "";
                this.Id_Testata = 0;
            }
        }
    }

}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/valutazioni/ValutazioneTestata.ts
import { ValutazioneTestataxAnno } from './ValutazioneTestataxAnno';
import { BaseCodeDescrStr } from '../baseClass/baseCodeDescrStr';

export type PKValutazioneTestata = typeof ValutazioneTestata.PK.prototype;
export class ValutazioneTestata {

    static PK = class {
        Piva: string;
        Id_Testata: number;

        constructor(piva: string, codice: number) {
            this.Piva = piva;
            this.Id_Testata = codice;
        }
    };

    primaryKey: PKValutazioneTestata;
    Ragione_Sociale: string;
    Valutazione_Piano_Cod: number;
    Valutazione_Piano_Des: string;
    Data_Redazione: string;
    Note : string;

    Valutazione_TestataxAnno: ValutazioneTestataxAnno[];

    flag_cancellazione: boolean;

    constructor(primaryKey: PKValutazioneTestata) {
        this.primaryKey = primaryKey;
        this.flag_cancellazione = false;
    }

}
